frontend/packages/console-shared/src/components/popover-body/popover-body.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Dropdown } from '@console/internal/components/utils/dropdown';
+import { Link } from 'react-router-dom';
+import { DataPoint } from '@console/internal/components/graphs';
+import './popoverBody.scss';
+import { ExternalLink, Humanize, resourcePathFromModel } from '@console/internal/components/utils';
+import { K8sKind } from '@console/internal/module/k8s';
+import { PodModel, NodeModel, ProjectModel } from '@console/internal/models';
+import { connectToURLs, MonitoringRoutes } from '@console/internal/reducers/monitoring';
+import { getPrometheusExpressionBrowserURL } from '@console/internal/components/graphs/prometheus-graph';
+
+const dropdownKeys = ['By Pods', 'By Node', 'By Projects'];
+const models = [PodModel, NodeModel, ProjectModel];
+const dropdownItems = _.zipObject(dropdownKeys, dropdownKeys);
+
+export const PopoverBody = connectToURLs(MonitoringRoutes.Prometheus)(
+  React.memo(
+    ({ topConsumers, title, humanize, query, urls }: React.PropsWithoutRef<PopoverBodyProps>) => {
+      const [selectedFilter, setSelectedFilter] = React.useState(dropdownKeys[0]);
+      const [bodyData, setBodyData] = React.useState(null);
+
+      const consumer = title
+        ? title.padStart(title.length + 1, ' ').padEnd(title.length + 2, ' ')
+        : ' ';
+      const selectedIndex = dropdownKeys.indexOf(selectedFilter);
+      const link = getPrometheusExpressionBrowserURL(urls, [`topk(20, ${query})`]);
+
+      React.useEffect(() => {
+        let data = topConsumers[selectedIndex];
+        if (data) {
+          data = data.map((elem) => Object.assign({}, elem, { y: humanize(elem.y).string }));
+          setBodyData(data);
+        }
+      }, [humanize, selectedFilter, selectedIndex, topConsumers]);
+
+      return (
+        <div className="ocs-popover">
+          <div className="ocs-popover__header">{`Top${consumer}consumers`}</div>
+          <Dropdown
+            buttonClassName="ocs-popover__dropdown"
+            items={dropdownItems}
+            onChange={setSelectedFilter}
+            selectedKey={selectedFilter}
+          />
+          <div className="ocs-popover-items">
+            <Items items={bodyData} model={models[selectedIndex]} query={query[selectedIndex]} />
+          </div>
+          <div className="ocs-popover__external-link">
+            <ExternalLink href={link} text="View more" />
+          </div>
+        </div>
+      );
+    },
+  ),
+);
+
+const Items: React.FC<ItemsProps> = ({ items, model }) => {
+  return items ? (
+    <>
+      {items.map((item) => {
+        const title = String(item.x);
+        return (
+          <>
+            <Link
+              key={String(item.x)}
```

Comment:
React `key` prop is declared as `string | number`, so we can do:

```suggestion
              key={item.x}
```

Reply:
But the type for x is `Date|string`

Reply:
Sorry, didn't know the exact type of `item.x`. Thanks for the update.